cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -154,17 +156,37 @@ func (w *KubeWaiter) SetTimeout(timeout time.Duration) {
 // WaitForStaticPodControlPlaneHashes blocks until it timeouts or gets a hash map for all components and their Static Pods
 func (w *KubeWaiter) WaitForStaticPodControlPlaneHashes(nodeName string) (map[string]string, error) {
 
-	var mirrorPodHashes map[string]string
-	err := wait.PollImmediate(constants.APICallRetryInterval, w.timeout, func() (bool, error) {
+	componentHash := ""
+	var err error
+	mirrorPodHashes := map[string]string{}
+	for _, component := range constants.MasterComponents {
+		err = wait.PollImmediate(constants.APICallRetryInterval, w.timeout, func() (bool, error) {
+			componentHash, err = getStaticPodSingleHash(w.client, nodeName, component)
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		mirrorPodHashes[component] = componentHash
```

Comment:
what happens if a component timeout occurs mid way through this iteration? I _assume_ the error on L164 will be ignored and the previous `componentHash` will be used, right?

Reply:
good catch, adding err checking logic when wait.PollImmediate returns err != nil stop processing and return err to the caller.